docs/src/basics/structs_and_tuples.md

```diff
@@ -41,6 +41,24 @@ _This information is not vital if you are new to the language, or programming in
 
 Structs have zero memory overhead. What that means is that in memory, each struct field is laid out sequentially. No metadata regarding the struct's name or other properties is preserved at runtime. In other words, structs are compile-time constructs. This is the same in Rust, but different in other languages with runtimes like Java.
 
+## Tuples
+
+Tuples are a [basic static-length type](./built_in_types.md#tuple-types) which contain multiple different types within themselves. The type of a type is defined by the values within it, and a tuple can contain basic types as well as structs and enums. An example of a tuple declaration is provided below.
+
+```sway
```

Comment:
I would prefer if new code snippets were put into `examples/`, but these are small so I could let it slide this time.

Reply:
Ok, will keep that in mind for future PRs. Snippets like this hopefully won't change much with future releases (declarations and simple operations) but I get why that could be harder to maintain on larger examples such as the example programs

Reply:
Keep in mind we can also use ranges to lift certain slices of code out of the `test` directory if we want to - could be a nice alternative for smaller cases where we want small snippets tested, but don't want to make a whole new example for a small snippet.

Reply:
That would be a good way to pull small snippets, would it also be possible then to pull code from the `std` library as it is filled with useful and high quality code?

Reply:
Yes, it should be safe to reference anything within this repo.

Here's an example of how a sub-snippet is [referenced](https://github.com/FuelLabs/sway/blob/master/docs/src/examples/subcurrency.md) in markdown, and here it is [rendered](https://fuellabs.github.io/sway/latest/examples/subcurrency.html).

Here, only the `body` section of the example is included, described by the postfix `:body` after the filename.

```
{{#include ../../../examples/subcurrency/src/main.sw:body}}
```

If we take a look at `examples/subcurrency/src/main.sw` we can see the "anchor" provided [here](https://github.com/FuelLabs/sway/blob/master/examples/subcurrency/src/main.sw#L1). Since this was written, the "body" anchor used has come to represent the entire file, but similarly anchors can be used to include smaller subsections of code.

I think the benefit of using comment anchors over line numbers is that, in the case that the code (example or test) is modified for whatever reason in the future, the snippet should still contain the relevant section. The risk with line numbers of course is that the slice slowly becomes offset from the desired code as the code is updated in the future. All that said, I'm not even sure it's possible to use direct line number ranges in the `#include` preprocessor anyway.

Reply:
Interesting, will try toying around with in my future docs PRs then so hopefully it is easier to maintain. I didn't know Sway could use Anchor's like that, so will use it as the default approach for include smaller subsections of code in `examples` for any future docs updates